engine/src/main/java/org/hibernate/search/query/engine/impl/QueryFilterset.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.query.engine.impl;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.BooleanClause.Occur;
+
+/**
+ * Internal representation for a set of filters to be applied on a query.
+ *
+ * @author Sanne Grinovero
+ */
+public final class QueryFilterset {
+
+	public static final QueryFilterset EMPTY_FILTERSET = new QueryFilterset();
+
+	private final List<Query> filterQueries;
```

Comment:
The name "...set" and this impl type (list) made me stumble. I'd align it, or better yet, change the class name into something more agnostic such as `FilteringQuery`.